This is working well but could use a tweak to get rid of a warning on compile. Maybe you can do it for me please  @dalto8 
` LuksBootKeyFileJob.cpp:246:1: warning: ‘bool hasEncryptedRoot()’ defined but not used [-Wunused-function]
  246 | hasEncryptedRoot()`

--- a/src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp
+++ b/src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp
@@ -242,13 +242,6 @@ hasUnencryptedSeparateBoot()
         { return hasMountPoint( partition, QStringLiteral( "/boot" ) ) && !isEncrypted( partition ); } );
 }
 
-STATICTEST bool
-hasEncryptedRoot()
-{
-    return anyPartition( []( const QVariantMap& partition )
-                         { return hasMountPoint( partition, QStringLiteral( "/" ) ) && isEncrypted( partition ); } );
-}
-
 Calamares::JobResult
 LuksBootKeyFileJob::exec()
 {
@@ -300,7 +293,7 @@ LuksBootKeyFileJob::exec()
         cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
         return Calamares::JobResult::ok();
     }
-    
+
     if ( s.devices.first().passphrase.isEmpty() )
     {
         cDebug() << Logger::SubEntry << "No root passphrase.";

--- a/src/modules/luksbootkeyfile/Tests.cpp
+++ b/src/modules/luksbootkeyfile/Tests.cpp
@@ -158,7 +158,6 @@ LuksBootKeyFileTests::testAnyPartition()
                             { return hasMountPoint( partdata, QStringLiteral( "/" ) ); } ) );
     QVERIFY( !anyPartition( []( const QVariantMap& partdata ) { return hasMountPoint( partdata, QString() ); } ) );
 
-    QVERIFY( !hasEncryptedRoot() );
     QVERIFY( hasUnencryptedSeparateBoot() );
 }
 

